### DISCUSSION TITLE
Redirect URI question with Tidal Login Service

### DISCUSSION THREAD

**Discussion by davddavd** [tidal-music/discussions#100] [Q&A] [answered]
Hi,

I have a working authorization code implementation but I get an error message following the the Tidal Login page using Firefox ESR.  The webapi desktop app registers and listens for the redirect using a ServerSocket on the loopback address.

redirect_uri="http://localhost:8888/callback" and the 
authUri = "https://login.tidal.com/authorize" 
					+ "?response_type=code" 
					+ "&client_id=" + URLEncoder.encode(clientId, "UTF-8")
					+ "&redirect_uri=" + URLEncoder.encode(redirectUri, "UTF-8") 
					+ "&scope=" + URLEncoder.encode(scope, "UTF-8") 
					+ "&code_challenge="+ URLEncoder.encode(codeChallenge, "UTF-8") 
					+ "&code_challenge_method=S256" 
					+ "&state="+URLEncoder.encode(String.valueOf(state), "UTF-8");

waiter = new Thread(() -> waitForAuthorizationCode());
waiter.start();
if (Desktop.isDesktopSupported())
    Desktop.getDesktop().browse(new URI(authUri));

(waitForAuthorizationCode snippet)
...
serverSocket = new ServerSocket(port);	
clientSocket = serverSocket.accept();
...

The good news is that it works.  I get the redirect from the Tidal Login Service and the authorization code can be successfully exchanged for the access token exactly as planned.

My issue is that I get the message after the Tidal Login Page:

Unable to connect

Firefox can’t establish a connection to the server at localhost:8888.

    The site could be temporarily unavailable or too busy. Try again in a few moments.
    If you are unable to load any pages, check your computer’s network connection.
    If your computer or network is protected by a firewall or proxy, make sure that Firefox is permitted to access the web.

This is message misleading because it appears the login service is failing but it does not.  I understand that Firefox shouldn't
be connectiing to this address that way and may be a browser issue but to customers' this message is undesirable and needs
supression.   The login page does show an animated circle with a successfull check mark when the sign in works but it is immediately
cleared before the above can't establish connection message is displayed a split second later.

Thanks in advance for any feedback. Btw, I waited about a month before posting in case this was a temporary problem already.

**Comment by davddavd** [comment, marked as answer]
So sorry but I realized that I was getting a couple more redirects that I handled in error one of which was a favorite's icon request that comes in after the successful login.  oops

**Reply by aliaksei-taliuk** [reply]
@davddavd no worries & good luck with your experiments!
(closing the discussion)
